src/python/WMComponent/DBS3Buffer/Oracle/NewDataset.py

```diff
@@ -12,6 +12,13 @@
 
 class NewDataset(MySQLNewDataset):
 
-    sql = """INSERT INTO dbsbuffer_dataset (path, processing_ver, acquisition_era, valid_status, global_tag, parent, prep_id)
-               SELECT :path, :processing_ver, :acquisition_era, :valid_status, :global_tag, :parent, :prep_id FROM DUAL
-               WHERE NOT EXISTS (SELECT path FROM dbsbuffer_dataset WHERE path = :path)"""
+    sql = """INSERT INTO dbsbuffer_dataset
+             (id, path, processing_ver, acquisition_era, valid_status, global_tag, parent, prep_id)
+             SELECT dbsbuffer_dataset_seq.nextval, :path, :processing_ver, :acquisition_era,
```

Comment:
Not on Oracle. There you create a sequence, but you have to either use them in the insert or (what was done before here) define an Oracle trigger that does it for you on each insert.